.github/workflows/unit-testing-and-coverage.yml

```diff
@@ -23,63 +23,86 @@ jobs:
   qa_docs:
     runs-on: ubuntu-20.04
     steps:
-      - name: "Checkout latest commit"
+      - name: "Checkout latest PR commit"
         uses: actions/checkout@v2
-      - name: "Set up Python 3.8"
+      - name: "Set up Python"
         uses: actions/setup-python@v2
         with:
+          # 3.8 is chosen because it seems to be the fastest for <3.9
+          # (3.9 excluded because it seems to be still very unstable)
           python-version: 3.8
-      - name: "Install dependencies"
+      - name: "Update modules & install deps"
+        # language=bash
         run: |
-          python -m pip install --upgrade pip
-          pip install tox
-      - name: "Execute tox -e qa,docs"
+          python -m pip install --upgrade pip setuptools wheel
+          python setup.py develop
+      - name: "flake8 Style Checking"
+        # language=bash
         run: |
-          tox -e qa,docs
+          pip install colorama flake8
+          python -m flake8 aiosmtpd setup.py housekeep.py release.py
+      - name: "Docs Checking"
+        # language=bash
+        run: |
+          pip install colorama pytest pytest-mock sphinx sphinx-autofixture sphinx_rtd_theme
+          pytest -v aiosmtpd/docs
+          python setup.py build_sphinx
+      - name: "Static Type Checking"
+        # language=bash
+        run: |
+          pip install pytype
+          pytype --keep-going --jobs auto .
+      - name: "Other QA Checks"
+        # language=bash
+        run: |
+          pip install pytest pytest-mock check-manifest
+          git fetch --no-tags --prune --no-recurse-submodules --depth=1 origin master:master
```

Comment:
This line is needed because `actions/checkout@v2` pulls exactly commit, and the `master` branch -- needed by one of the qa tests -- was not created.